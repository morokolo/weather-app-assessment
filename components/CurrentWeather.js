import React from "react";
import { Image, ImageBackground, StyleSheet, Text, View } from "react-native";
import colors from "../constants/colors";
import bgImageDictionary from "../utils/bgImageDictionary";
import iconDictionary from "../utils/iconDictionary";
import HourlyWeather from "./HourlyWeather";

export default function CurrentWeather({ forecast, current }) {
  return (
    <ImageBackground
      blurRadius={2}
      resizeMode="contain"
      style={{
        flex: 1,
      }}
      source={bgImageDictionary[current.icon]}
    >
      <View
        style={{
          flex: 1,
          justifyContent: "center",
          alignItems: "center",
        }}
      >
        <Text style={styles.currentTemp} testID="text">
          {Math.round(forecast.current.temp)}Â°
        </Text>
        <Image
          style={styles.temperatureIndicatorIcon}
          source={iconDictionary[current.icon] || iconDictionary["01d"]}
        />
        <Text style={styles.currentDescription}>{current.description}</Text>
      </View>
      <HourlyWeather forecast={forecast} />
    </ImageBackground>
  );
}

const styles = StyleSheet.create({
  currentTemp: {
    fontSize: 50,
    textAlign: "center",
    color: colors.white,
    fontWeight: "600",
  },
  currentDescription: {
    width: "100%",
    textAlign: "center",
    fontWeight: "500",
    fontSize: 20,
    textTransform: "uppercase",
    color: colors.white,
  },

  temperatureIndicatorIcon: {
    width: 40,
    height: 40,
    alignItems: "center",
  },
});
