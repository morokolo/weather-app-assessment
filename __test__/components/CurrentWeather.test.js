import "react-native";
import React from "react";
import CurrentWeather from "../../components/CurrentWeather";
import renderer from "react-test-renderer";

import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

let current = {
  description: "clear sky",
  icon: "01n",
  id: 800,
  main: "Clear",
};

let forecast = {
  current: {
    dt: 1614065857,
    sunrise: 1614005421,
    sunset: 1614045368,
    temp: 12.58,
    feels_like: 10.8,
    pressure: 1022,
    humidity: 71,
    dew_point: 7.47,
    uvi: 0,
    clouds: 20,
    visibility: 10000,
    wind_speed: 1.7,
    wind_deg: 12,
    weather: [
      {
        id: 801,
        main: "Clouds",
        description: "few clouds",
        icon: "02n",
      },
    ],
  },
  hourly: [
    {
      dt: 1614063600,
      temp: 12.58,
      weather: [
        {
          id: 801,
          main: "Clouds",
          description: "few clouds",
          icon: "02n",
        },
      ],
    },
  ],
};

it("should render the App component with  send props", () => {
  const tree = renderer
    .create(<CurrentWeather forecast={forecast} current={current} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("should render children, containing grand children", () => {
  const tree = renderer
    .create(<CurrentWeather forecast={forecast} current={current} />)
    .toJSON();

  expect(tree.children.length).toBe(3);
});

it("renders a Image element type", () => {
  const tree = renderer
    .create(<CurrentWeather forecast={forecast} current={current} />)
    .toJSON();

  expect(tree.children[0].type).toBe("Image");
});

it("renders a Image element type", () => {
  const tree = renderer
    .create(<CurrentWeather forecast={forecast} current={current} />)
    .toJSON();

  expect(tree.children[2].type).toBe("View");
});

it("renders a Image element type", () => {
  const tree = renderer
    .create(<CurrentWeather forecast={forecast} current={current} />)
    .toJSON();

  expect(tree.children[2].type).toBe("View");
});
