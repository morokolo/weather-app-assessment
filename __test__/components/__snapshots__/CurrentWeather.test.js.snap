// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the App component with  send props 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Image
    blurRadius={2}
    resizeMode="contain"
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 50,
          "fontWeight": "600",
          "textAlign": "center",
        }
      }
      testID="text"
    >
      13
      °
    </Text>
    <Image
      source={1}
      style={
        Object {
          "alignItems": "center",
          "height": 40,
          "width": 40,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 20,
          "fontWeight": "500",
          "textAlign": "center",
          "textTransform": "uppercase",
          "width": "100%",
        }
      }
    >
      clear sky
    </Text>
  </View>
  <View>
    <RCTScrollView
      data={
        Array [
          Object {
            "dt": 1614063600,
            "temp": 12.58,
            "weather": Array [
              Object {
                "description": "few clouds",
                "icon": "02n",
                "id": 801,
                "main": "Clouds",
              },
            ],
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={true}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#57575da6",
                "padding": 6,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontWeight": "bold",
                }
              }
            >
              09:00:00
            </Text>
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontWeight": "bold",
                }
              }
            >
              13
              °C
            </Text>
            <Image
              source={1}
              style={
                Object {
                  "height": 30,
                  "marginBottom": 35,
                  "width": 30,
                }
              }
            />
            <Text>
              few clouds
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
